source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,931 @@
+    buffCfg[ 0 ].txBdStartAddrAlign =
+        &( txBuffDescrip_0[ 0 ] );                            /* Aligned transmit buffer descriptor start address. */
+    buffCfg[ 0 ].rxBufferAlign =
+        &( rxDataBuff_0[ 0 ][ 0 ] );                          /* Receive data buffer start address. */
+    buffCfg[ 0 ].txBufferAlign = &( txDataBuff_0[ 0 ][ 0 ] ); /* Transmit data buffer start address. */
+    buffCfg[ 0 ].txFrameInfo = NULL;                          /* Transmit frame information start address. Set only if using zero-copy transmit. */
+    buffCfg[ 0 ].rxMaintainEnable = true;                     /* Receive buffer cache maintain. */
+    buffCfg[ 0 ].txMaintainEnable = true;                     /* Transmit buffer cache maintain. */
+
+    sysClock = phyHandle.mdioHandle->resource.csrClock_Hz;
+
+    ENET_GetDefaultConfig( &config );
+
+    config.ringNum = ENET_RING_NUM;
+    config.rxBuffAlloc = NULL;
+    config.rxBuffFree = NULL;
+    config.userData = ethernetifLocal;
+    config.miiSpeed = ( enet_mii_speed_t ) speed;
+    config.miiDuplex = ( enet_mii_duplex_t ) duplex;
+
+    /* Only get interrupt for incoming messages. */
+    config.interrupt = kENET_RxFrameInterrupt;
+    config.callback = ethernet_callback;
+
+    for( instance = 0; instance < ARRAY_SIZE( enetBases ); instance++ )
+    {
+        if( enetBases[ instance ] == ethernetifLocal->base )
+        {
+            NVIC_SetPriority( enetRxIrqId[ instance ], ENET_PRIORITY );
+            NVIC_SetPriority( enetTxIrqId[ instance ], ENET_PRIORITY );
+            break;
+        }
+    }
+
+    configASSERT( instance != ARRAY_SIZE( enetBases ) );
+
+    if( instance == ARRAY_SIZE( enetBases ) )
+    {
+        xStatus = kStatus_Fail;
+    }
+    else
+    {
+        pxEndPoint = FreeRTOS_FirstEndPoint( pxMyInterface );
+        configASSERT( pxEndPoint != NULL );
+
+        for( i = 0; i < ENET_RXBUFF_NUM; i++ )
+        {
+            ethernetifLocal->RxPbufs[ i ].buffer = &( ethernetifLocal->RxDataBuff[ i ][ 0 ] );
+            ethernetifLocal->RxPbufs[ i ].buffer_used = false;
+        }
+
+        /* Initialize the ENET module. */
+        xStatus = ENET_Init( ethernetifLocal->base,
+                             &ethernetifLocal->handle,
+                             &config,
+                             &buffCfg[ 0 ],
+                             pxEndPoint->xMACAddress.ucBytes,
+                             sysClock );
+
+        #if ( ipconfigUSE_LLMNR == 1 )
+            ENET_AddMulticastGroup( ethernetifLocal->base, ( uint8_t * ) xLLMNR_MacAdress.ucBytes );
+        #endif /* ipconfigUSE_LLMNR */
+
+        #if ( ipconfigUSE_IPv6 != 0 )
+            #if ( ipconfigUSE_LLMNR == 1 )
+                ENET_AddMulticastGroup( ethernetifLocal->base, ( uint8_t * ) xLLMNR_MacAdressIPv6.ucBytes );
+            #endif /* ipconfigUSE_LLMNR */
+
+            for( pxEndPoint = FreeRTOS_FirstEndPoint( pxMyInterface );
```

Comment:
Is all nodes IPv6 multicast MAC address requires to be added to the list as well? Or is already allowed?

Reply:
Sadly I don't have set-up to test IPv6 locally. Added description saying only IPv4 was verified.